libr/core/canal.c

```diff
@@ -767,6 +767,21 @@ static int core_anal_fcn(RCore *core, ut64 at, ut64 from, int reftype, int depth
 						fcnpfx = r_config_get (core->config, "anal.fcnprefix");
 					}
 					fcn->name = r_str_newf ("%s.%08"PFMT64x, fcnpfx, fcn->addr);
+					/* Import trampoline naming */
+					if (r_list_length (fcn->bbs) == 1
```

Comment:
too many nested `ifs` imo

Reply:
is this necessary? Can't check whether the function has a single JMP instruction and that's it? number of instructions of a function should be available to you